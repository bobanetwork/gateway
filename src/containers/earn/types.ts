export enum toLayer {
  L1 = 'L2',
  L2 = 'L1',
}
