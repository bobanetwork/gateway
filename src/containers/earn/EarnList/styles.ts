import styled from 'styled-components'

export const EarnListContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px 0;
  padding: 10px 0;
`
