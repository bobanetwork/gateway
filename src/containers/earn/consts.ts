export const tableHeaderOptions = [
  { name: 'Token', width: 225 },
  {
    name: 'Available Balance',
    tooltip:
      'Available Balance refers to the amount of funds currently in each pool.',
    width: 145,
  },
  {
    name: 'Total Staked',
    tooltip: 'Total staked denotes the funds staked by liquidity providers.',
    width: 115,
  },
  {
    name: 'APR',
    tooltip:
      'The APR is the historical APR, which reflects the fees people paid to bridge and the previous usage patterns for each pool.',
    width: 85,
  },
  { name: 'Your Stake', width: 90 },
  { name: 'Earned', width: 110 },
  { name: 'Actions', width: 75 },
]
