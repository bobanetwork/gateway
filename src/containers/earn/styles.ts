import styled, { css } from 'styled-components'
import { Typography } from 'components/global/typography'

export const EarnPageContainer = styled.div`
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: 10px;
  padding-top: 0;
  width: 1025px;
`

export const LayerAlert = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 30px;
  border-radius: 8px;
  margin: 8px 0;
  padding: 10px;
  background: ${(props) => props.theme.colors.box.background};
`

export const Help = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 30px;
  margin: 10px 0;
  padding: 10px;
  border-radius: 8px;
  background: ${(props) => props.theme.colors.box.background};
  ${({ theme: { name, colors } }) => css`
    border: 1px solid ${name === 'light' ? colors.blue[200] : colors.blue[100]};
    background: ${name === 'light' ? colors.blue[50] : colors.blue[500]};
    color: ${name === 'light' ? colors.blue[500] : colors.blue[100]};
  `}
`

export const AlertText = styled(Typography)`
  margin-left: 10px;
  flex: 4;
`

export const AlertInfo = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex: 1;
`

export const EarnAction = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
`

export const EarnActionContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 10px 0;
`

export const EarnListContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px 0;
  padding: 10px 0;
`

export const TabSwitcherContainer = styled.div`
  display: flex;
  padding: 4px;
  gap: 0 15px;
  border-radius: 8px;
  ${({ theme: { colors, name } }) =>
    name === 'light'
      ? css`
          background: ${colors.gray[50]};
        `
      : css`
          background: ${colors.gray[500]};
        `}
`

export const Tab = styled.div<{ active: boolean }>`
  padding: 8px 24px;
  border-radius: 8px;
  cursor: pointer;
  ${(props) =>
    props.active &&
    `
        color:${props.theme.colors.gray[800]};
        background:${props.theme.colors.green[300]}
    `}
`
