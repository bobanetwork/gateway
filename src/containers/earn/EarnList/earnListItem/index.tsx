import { getEarnInfo, updateWithdrawToken, getReward } from 'actions/earnAction'
import { openAlert, openModal } from 'actions/uiAction'
import { IconLabel, TableContent, Typography } from 'components/global'
import {
  StyledLabel,
  StyledMenu,
  StyledMenuItem,
} from 'components/global/menu/styles'
import { BigNumber, BigNumberish } from 'ethers'
import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import {
  selectAccountEnabled,
  selectlayer1Balance,
  selectlayer2Balance,
} from 'selectors'
import networkService from 'services/networkService'
import { LiquidityPoolLayer } from 'types/earn.types'
import { formatLargeNumber, logAmount, powAmount } from 'util/amountConvert'
import { EarnListItemContainer, IconWrapper, MoreActionIcon } from './styles'

interface EarnListItemProps {
  userInfo: any
  poolInfo: any
  chainId?: any
  tokenAddress: string
  lpChoice: LiquidityPoolLayer
  showMyStakeOnly: boolean
}

const EarnListItem = ({
  poolInfo,
  userInfo,
  chainId,
  tokenAddress,
  lpChoice,
  showMyStakeOnly = false,
}: EarnListItemProps) => {
  const dispatch = useDispatch<any>()

  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)

  const accountEnabled = useSelector(selectAccountEnabled())
  const layer1Balance = useSelector(selectlayer1Balance)
  const layer2Balance = useSelector(selectlayer2Balance)

  const open = Boolean(anchorEl)
  const symbol = poolInfo.symbol
  const name = poolInfo.name
  const decimals = poolInfo.decimals
  const disabled = !lpChoice.includes(networkService.L1orL2!)
  const address =
    lpChoice === LiquidityPoolLayer.L1LP
      ? poolInfo.l1TokenAddress
      : poolInfo.l2TokenAddress

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget)
  }

  const handleClose = () => {
    setAnchorEl(null)
  }

  const formatNumber = (value: BigNumberish, limit?: any) => {
    const limits = limit || 2
    return formatLargeNumber(Number(logAmount(value, decimals, limits)))
  }

  let userReward: BigNumberish = 0

  if (
    Object.keys(userInfo).length &&
    Object.keys(poolInfo).length &&
    accountEnabled
  ) {
    userReward = BigNumber.from(userInfo.pendingReward)
      .add(
        BigNumber.from(userInfo.amount)
          .mul(BigNumber.from(poolInfo.accUserRewardPerShare))
          .div(BigNumber.from(powAmount(1, 12)))
          .sub(BigNumber.from(userInfo.rewardDebt))
      )
      .toString()
  }

  const getBalance = (address: string) => {
    let tokens = []
    if (lpChoice.includes('L1')) {
      tokens = Object.values(layer1Balance)
    } else if (lpChoice.includes('L2')) {
      tokens = Object.values(layer2Balance)
    }
    const token: any = tokens.find(
      (t: any) => t.address.toLowerCase() === address.toLowerCase()
    )
    return token ? [token.balance, token.decimals] : [0, 0]
  }

  const handleHarvest = async () => {
    const rewardTx = await dispatch(
      getReward(
        lpChoice === LiquidityPoolLayer.L1LP
          ? poolInfo.l1TokenAddress
          : poolInfo.l2TokenAddress,
        userReward,
        lpChoice as any
      )
    )

    if (rewardTx) {
      dispatch(
        openAlert(
          `${logAmount(userReward, poolInfo.decimals, 2)} ${
            poolInfo.symbol
          } was added to your account`
        )
      )
      dispatch(getEarnInfo())
    }
  }

  const handleUnstakeToken = () => {
    let currency = poolInfo.l1TokenAddress
    let LPAddress = poolInfo.L1LPAddress
    if (lpChoice === LiquidityPoolLayer.L2LP) {
      currency = poolInfo.l2TokenAddress
      LPAddress = poolInfo.L2LPAddress
    }

    dispatch(
      updateWithdrawToken({
        symbol: poolInfo.symbol,
        currency,
        LPAddress,
        L1orL2Pool: lpChoice,
        balance: getBalance(tokenAddress)[0],
        decimals: poolInfo.decimals,
      })
    )
    dispatch(openModal('EarnWithdrawModal'))
  }

  let enableReward = false
  if (Number(logAmount(userReward, decimals, 3)) >= 0.001) {
    enableReward = true
  }

  const actionItems = [
    {
      disabled,
      label: 'Unstake',
      onClick: () => {
        handleUnstakeToken()
      },
    },
    {
      label: 'Harvest',
      disabled:
        disabled || logAmount(userReward, decimals) === '0' || !enableReward,
      onClick: () => {
        console.log(`harvest token`)
        handleHarvest()
      },
    },
  ]

  const tableOptions = [
    {
      content: (
        <IconLabel token={{ name, symbol, address, chainId, decimals }} />
      ),
      width: 225,
    },
    {
      content: (
        <Typography variant="body2">
          {formatNumber(poolInfo.tokenBalance)}
        </Typography>
      ),
      width: 145,
    },
    {
      content: (
        <Typography variant="body2">
          {formatNumber(poolInfo.userDepositAmount)}
        </Typography>
      ),
      width: 115,
    },
    {
      content: (
        <Typography variant="body2">{`${logAmount(
          poolInfo.APR,
          0,
          2
        )}`}</Typography>
      ),
      width: 85,
    },
    {
      content: (
        <Typography variant="body2">
          {userInfo.amount ? `${logAmount(userInfo.amount, decimals, 2)}` : `0`}
        </Typography>
      ),
      width: 90,
    },
    {
      content: (
        <Typography variant="body2">
          {userReward ? `${logAmount(userReward, decimals, 5)}` : `0`}
        </Typography>
      ),
      width: 110,
    },
    {
      content: (
        <>
          <IconWrapper onClick={handleClick}>
            <MoreActionIcon />
          </IconWrapper>
          <StyledMenu
            anchorEl={anchorEl}
            id={name}
            open={open}
            onClose={handleClose}
            onClick={handleClose}
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'center',
            }}
            transformOrigin={{
              vertical: 'top',
              horizontal: 'center',
            }}
          >
            {actionItems.map((opt, index) => {
              return (
                <StyledMenuItem
                  key={index}
                  disabled={opt.disabled}
                  onClick={opt.onClick}
                >
                  <StyledLabel>{opt.label}</StyledLabel>
                </StyledMenuItem>
              )
            })}
          </StyledMenu>
        </>
      ),
      width: 75,
    },
  ]

  if (showMyStakeOnly === true) {
    if (Number(logAmount(userInfo.amount, decimals, 2)) < 0.001) {
      return null
    }
  }

  return (
    <EarnListItemContainer>
      <TableContent options={tableOptions} />
    </EarnListItemContainer>
  )
}

export default React.memo(EarnListItem)
