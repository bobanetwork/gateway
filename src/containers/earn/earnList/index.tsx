import { TableHeader } from 'components/global/table'
import React, { Fragment } from 'react'
import { useSelector } from 'react-redux'
import { selectPoolInfo, selectUserInfo } from 'selectors'
import { EarnListProps } from 'types/earn.types'
import { tableHeaderOptions } from '../table.header'
import EarnListItem from './earnListItem'
import { EarnListContainer, EearnListLoadingContainer } from './styles'
import { CircularProgress } from '@mui/material'

const EarnList = ({ lpChoice, showMyStakeOnly }: EarnListProps) => {
  const userInfo = useSelector(selectUserInfo())
  const poolInfo = useSelector(selectPoolInfo())
  const poolList: string[] = Object.keys(poolInfo[lpChoice])

  const isLoading = poolList.filter(Boolean).length === 0

  return (
    <>
      <TableHeader options={tableHeaderOptions} />
      {isLoading && (
        <EearnListLoadingContainer>
          <CircularProgress color="secondary" />
        </EearnListLoadingContainer>
      )}
      {!isLoading && (
        <>
          <EarnListContainer>
            {poolList.map((tokenAddress, i) => {
              return (
                <Fragment key={i}>
                  <EarnListItem
                    showMyStakeOnly={showMyStakeOnly}
                    poolInfo={poolInfo[lpChoice][tokenAddress]}
                    userInfo={userInfo[lpChoice][tokenAddress]}
                    lpChoice={lpChoice}
                    tokenAddress={tokenAddress}
                  />
                </Fragment>
              )
            })}
          </EarnListContainer>
        </>
      )}
    </>
  )
}

export default React.memo(EarnList)
