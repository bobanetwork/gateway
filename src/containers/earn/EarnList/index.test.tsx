import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import CustomThemeProvider from 'themes'
import { mockedInitialState } from 'util/tests'
import EarnList, { EarnListProps } from '.'
import { LiquidityPoolLayer } from 'types/earn.types'

const mockStore = configureStore([thunk])

const renderEarnComponent = ({ lpChoice, showMyStakeOnly }: EarnListProps) => {
  return render(
    <Provider store={mockStore(mockedInitialState)}>
      <BrowserRouter>
        <CustomThemeProvider>
          <EarnList lpChoice={lpChoice} showMyStakeOnly={showMyStakeOnly} />
        </CustomThemeProvider>
      </BrowserRouter>
    </Provider>
  )
}

describe('EarnList ', () => {
  test('should match snapshot in case no data', () => {
    const { asFragment } = renderEarnComponent({
      lpChoice: LiquidityPoolLayer.L1LP,
      showMyStakeOnly: false,
    })
    expect(asFragment).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
  xtest('should match snapshot with data', () => {
    // TODO: updated the initial state with static data to feed to earn list.
    const { asFragment } = renderEarnComponent({
      lpChoice: LiquidityPoolLayer.L1LP,
      showMyStakeOnly: false,
    })
    expect(asFragment).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
})
