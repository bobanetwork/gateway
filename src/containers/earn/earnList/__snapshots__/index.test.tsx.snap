// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EarnList  should match snapshot in case no data 1`] = `
<DocumentFragment>
  <div
    class="sc-fqkvVR fuyRCw sc-iGgWBj cCxFlv"
  >
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 225px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Token
      </p>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 145px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Available Balance
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-kud2p2-MuiSvgIcon-root"
        data-mui-internal-clone-element="true"
        data-testid="HelpOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
        />
      </svg>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 115px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Total Staked
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-kud2p2-MuiSvgIcon-root"
        data-mui-internal-clone-element="true"
        data-testid="HelpOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
        />
      </svg>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 85px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        APR
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-kud2p2-MuiSvgIcon-root"
        data-mui-internal-clone-element="true"
        data-testid="HelpOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
        />
      </svg>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 90px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Your Stake
      </p>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 110px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Earned
      </p>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 75px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Actions
      </p>
    </div>
  </div>
  <div
    class="sc-fHjqPf kAuoON"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorSecondary css-1szo49z-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`EarnList  should match snapshot with data 1`] = `
<DocumentFragment>
  <div
    class="sc-fqkvVR fuyRCw sc-iGgWBj cCxFlv"
  >
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 225px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Token
      </p>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 145px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Available Balance
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-kud2p2-MuiSvgIcon-root"
        data-mui-internal-clone-element="true"
        data-testid="HelpOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
        />
      </svg>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 115px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Total Staked
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-kud2p2-MuiSvgIcon-root"
        data-mui-internal-clone-element="true"
        data-testid="HelpOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
        />
      </svg>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 85px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        APR
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-kud2p2-MuiSvgIcon-root"
        data-mui-internal-clone-element="true"
        data-testid="HelpOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
        />
      </svg>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 90px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Your Stake
      </p>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 110px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Earned
      </p>
    </div>
    <div
      class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
      style="max-width: 75px;"
    >
      <p
        class="sc-gEvEer hiDcmV sc-eqUAAy jSVBRG"
      >
        Actions
      </p>
    </div>
  </div>
  <div
    class="sc-hzhJZQ jXgdnO"
  >
    <div
      class="sc-dAbbOL kfujas"
      data-testid="earn-list-item"
    >
      <div
        class="sc-fqkvVR fuyRCw sc-gsFSXq djGaQM"
      >
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 225px;"
        >
          <div
            class="sc-fqkvVR fuyRCw sc-kpDqfm fYCSlX"
          >
            <div
              class="sc-dcJsrY jqXhqy"
            >
              <div
                class="sc-fqkvVR fuyRCw"
              >
                <div>
                  <img
                    alt="logo"
                    class="sc-dhKdcB ldcSKy"
                    src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/boba.svg"
                  />
                </div>
                <div>
                  <p
                    class="sc-imWYAI lgSgkc"
                  >
                    BOBA
                  </p>
                  <p
                    class="sc-jXbUNg iWjFhm"
                  >
                    Boba Token
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 145px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            99.95k
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 115px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            100.00k
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 85px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0.00
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 90px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 110px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 75px;"
        >
          <div
            class="sc-feUZmu ceIvoQ"
            data-testid="action-btn"
          >
            <div
              class="sc-fUnMCh euYGlO"
            >
              <div>
                <div
                  data-src="actions.svg"
                  src=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-dAbbOL kfujas"
      data-testid="earn-list-item"
    >
      <div
        class="sc-fqkvVR fuyRCw sc-gsFSXq djGaQM"
      >
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 225px;"
        >
          <div
            class="sc-fqkvVR fuyRCw sc-kpDqfm fYCSlX"
          >
            <div
              class="sc-dcJsrY jqXhqy"
            >
              <div
                class="sc-fqkvVR fuyRCw"
              >
                <div>
                  <img
                    alt="logo"
                    class="sc-dhKdcB ldcSKy"
                    src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/eth.svg"
                  />
                </div>
                <div>
                  <p
                    class="sc-imWYAI lgSgkc"
                  >
                    ETH
                  </p>
                  <p
                    class="sc-jXbUNg iWjFhm"
                  >
                    ETH
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 145px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0.85
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 115px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0.61
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 85px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0.10
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 90px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 110px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 75px;"
        >
          <div
            class="sc-feUZmu ceIvoQ"
            data-testid="action-btn"
          >
            <div
              class="sc-fUnMCh euYGlO"
            >
              <div>
                <div
                  data-src="actions.svg"
                  src=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-dAbbOL kfujas"
      data-testid="earn-list-item"
    >
      <div
        class="sc-fqkvVR fuyRCw sc-gsFSXq djGaQM"
      >
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 225px;"
        >
          <div
            class="sc-fqkvVR fuyRCw sc-kpDqfm fYCSlX"
          >
            <div
              class="sc-dcJsrY jqXhqy"
            >
              <div
                class="sc-fqkvVR fuyRCw"
              >
                <div>
                  <img
                    alt="logo"
                    class="sc-dhKdcB ldcSKy"
                    src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/omg.svg"
                  />
                </div>
                <div>
                  <p
                    class="sc-imWYAI lgSgkc"
                  >
                    OMG
                  </p>
                  <p
                    class="sc-jXbUNg iWjFhm"
                  >
                    OMG Token
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 145px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 115px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 85px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0.00
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 90px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 110px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 75px;"
        >
          <div
            class="sc-feUZmu ceIvoQ"
            data-testid="action-btn"
          >
            <div
              class="sc-fUnMCh euYGlO"
            >
              <div>
                <div
                  data-src="actions.svg"
                  src=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-dAbbOL kfujas"
      data-testid="earn-list-item"
    >
      <div
        class="sc-fqkvVR fuyRCw sc-gsFSXq djGaQM"
      >
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 225px;"
        >
          <div
            class="sc-fqkvVR fuyRCw sc-kpDqfm fYCSlX"
          >
            <div
              class="sc-dcJsrY jqXhqy"
            >
              <div
                class="sc-fqkvVR fuyRCw"
              >
                <div>
                  <img
                    alt="logo"
                    class="sc-dhKdcB ldcSKy"
                    src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/usdc.svg"
                  />
                </div>
                <div>
                  <p
                    class="sc-imWYAI lgSgkc"
                  >
                    USDC
                  </p>
                  <p
                    class="sc-jXbUNg iWjFhm"
                  >
                    USD Coin
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 145px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 115px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 85px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0.00
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 90px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 110px;"
        >
          <p
            class="sc-gEvEer hiDcmV"
          >
            0
          </p>
        </div>
        <div
          class="sc-fqkvVR fuyRCw sc-kAyceB xJAbn"
          style="max-width: 75px;"
        >
          <div
            class="sc-feUZmu ceIvoQ"
            data-testid="action-btn"
          >
            <div
              class="sc-fUnMCh euYGlO"
            >
              <div>
                <div
                  data-src="actions.svg"
                  src=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
